presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -213,6 +216,53 @@ public Page getNextPage()
         }
     }
 
+    private Block buildNullBlock(Type type, int batchSize)
+    {
+        BlockBuilder blockBuilder = type.createBlockBuilder(new BlockBuilderStatus(), batchSize);
+        for (int j = 0; j < batchSize; j++) {
+            blockBuilder.appendNull();
+        }
+        return blockBuilder.build();
+    }
+
+    private Block buildRowBlock(List<TypeSignatureParameter> parameters, int batchSize, List<String> path)
+    {
+        Block[] blocks = new Block[parameters.size()];
+        List<String> fieldNames = new ArrayList<>();
+        for (int i = 0; i < parameters.size(); i++) {
+            NamedTypeSignature namedTypeSignature = parameters.get(i).getNamedTypeSignature();
+            Type fieldType = typeManager.getType(namedTypeSignature.getTypeSignature());
+            String name = namedTypeSignature.getName();
+            path.add(name);
+            RichColumnDescriptor columnDescriptor = getDescriptor(fileSchema, requestedSchema, path);
+            if (columnDescriptor == null) {
+                path.remove(name);
+                blocks[i] = buildNullBlock(fieldType, batchSize);
+                continue;
+            }
+
+            if (ROW.equals(fieldType.getTypeSignature().getBase())) {
+                blocks[i] = buildRowBlock(fieldType.getTypeSignature().getParameters(), batchSize, path);
+            }
+            else {
+                try {
+                    blocks[i] = parquetReader.readBlock(columnDescriptor, fieldType);
```

Comment:
No LazyBlock for Struct's sub-blocks, only LazyBlock for first level columns
Only scanning necessary sub-blocks for requested nest struct is done via nested column pruning: 
https://github.com/prestodb/presto/pull/5547